src/components/RandomPresent/index.tsx

```diff
@@ -0,0 +1,65 @@
+import { LeftStep, useTypedTranslation, useInterval, RangePrice, type StylePropType, useScreenWidth } from "@shared";
+import { useEffect, useState } from "react";
+import { useGetRandomProductsQuery } from "@src/app/api/products";
+import { getLeft, getSize, getDirection, getStyle, getPresent } from "./helpers";
+import { useAppDispatch, useAppSelector } from "@src/app/store";
+import { setPermition } from "@src/app/store/slices/randomRange";
+
+export default function RandomPresent (): JSX.Element {
+  const randomRange = useAppSelector((state) => state.randomRange);
+  const { from, to } = randomRange;
+  const { data, isLoading, error } = useGetRandomProductsQuery({ from, to });
+  const t = useTypedTranslation();
+  const [random, setRandom] = useState(false);
+  const [r, setR] = useState(false);
+  const windowWidth = useScreenWidth();
+  const dispatch = useAppDispatch();
+
+  const [style, setStyle] = useState<StylePropType[]>([
+    { left: 5, size: 21, direction: "forward" },
+    { left: 20, size: 23, direction: "forward" },
+    { left: 37, size: 27, direction: "forward" },
+    { left: 58, size: 23, direction: "forward" },
+    { left: 75, size: 21, direction: "back" },
+  ]);
+
+  useInterval(() => {
+    if (random) {
+      setStyle(style.map((el) => ({
```

Comment:
If u want to update state itself, it is better to use callback function instead of direct state update. Especially since this is interval.
Read more: https://hello-js.com/articles/use-state-direct-value-or-callback-function/